radix-engine/src/model/royalty/module.rs

```diff
@@ -34,6 +34,24 @@ impl Default for RoyaltyModule {
     }
 }
 
+macro_rules! pre_load_vault {
```

Comment:
Can we add TODO to remove this at some point, definitely a code smell I think that we require this.

Reply:
Yes!